libhirte/src/orchestrator.c

```diff
@@ -69,14 +69,27 @@ Orchestrator *orch_new(const OrchestratorParams *params) {
                 return NULL;
         }
 
+        _cleanup_free_ char *service_name = NULL;
+        r = asprintf(&service_name, "%s", bus_service_name);
+        if (r < 0) {
+                fprintf(stderr, "Out of memory\n");
```

Comment:
In general I agree. But `libhirte` is a library and usually I wouldn't terminate a program here. I am uncertain how to properly handle errors in C since only the error codes convey to the caller what happened - and at the same time these codes are quite confusing.

Reply:
ack

Reply:
I recommend using assert() the same way as in systemd.  For ```asprintf()```, systemd has examples such as logging that an out of memory condition has occurred.  systemd uses ```assert()``` a lot to check for NULL for function arguments that are pointers.  In general, the purpose of ```assert()```, whether in a binary or library, is to check for an unrecoverable error.  In that situation it's better to go ahead and stop the program in order to avoid glossing over a bug that would otherwise be very difficult to debug.  For example, if you have a memory corruption bug, an assert might fail, and that then serves as important clue as to the source of the problem.  In my opinion, ```assert()``` is a good idea (and perhaps especially) even in mission critical systems.